tls/s2n_connection_serialize.c

```diff
@@ -119,6 +120,10 @@ int s2n_connection_serialize(struct s2n_connection *conn, uint8_t *buffer, uint3
         POSIX_GUARD_RESULT(s2n_connection_serialize_secrets(conn, &output));
     }
 
+    /* Users should not be able to use the connection after serialization as that
+     * could lead to nonce reuse. */
+    POSIX_GUARD_RESULT(s2n_connection_set_closed(conn));
```

Comment:
Actually, I think you could still send one more record on a closed connection: the close_notify alert. We might need a stronger defense.

And whatever we do, wiping the keys / ivs / wherever this data is stored is probably the safest. That just won't give good error messages, so should probably just be a defense in depth option.

Maybe: set closed, wipe keys, set client/server back to initial rather than secure. Test that if s2n_shutdown is called, the close_notify is sent in plaintext.

Reply:
Just seting closed + setting client/server back to initial should be sufficient. Actually wiping the ivs is just my paranoia in case we mess up or I'm missing another case.

Reply:
I think wiping as much as possible is a good defense-in-depth approach.